### PR TITLE
[tests-only] cache ocis build artifact for testing

### PR DESCRIPTION
**Pull request by wkloucek**
oCIS binary is build once and then provided for all tests (acceptance test, ui tests, but not for go linting, ...).

Diffs between master and branch look good for following tests:
- drone starlark
- drone starlark --build.event cron
- drone starlark --build.event pull
- drone starlark --build.event pull --build.message "[docs-only..]"
- drone starlark --build.message "[docs-only..]"

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -150,7 +153,7 @@ def main(ctx):
 
     pipelines = before + [ notify_pipeline ]
 
-  elif '[docs-only]' in (ctx.build.title + ctx.build.message):
+  elif '[docs-only]' in ctx.build.title:
```
Comment (wkloucek) [inline .drone.star:156]: PR messages mentioning [docs-only] should also be capable of running a full CI run. This eg. broke this PR completely, so I changed it.

File: .drone.star [hunk 2]
```diff
@@ -150,7 +153,12 @@ def main(ctx):
 
     pipelines = before + [ notify_pipeline ]
 
-  elif '[docs-only]' in (ctx.build.title + ctx.build.message):
+  elif \
+  (ctx.build.event == "pull" and '[docs-only]' in ctx.build.title) \
+  or \
+  (ctx.build.event != "pull" and '[docs-only]' in (ctx.build.title + ctx.build.message)):
+  # [docs-only] is not taken from PR messages, but from commit messages
+
```
Comment (wkloucek) [inline .drone.star:161]: a PR mentioning [docs-only] in its message should also run a full CI run. This odd behaviour caused some confusion for me on this PR...

**Review by phil-davis — APPROVED** [review]
Looks good - the cache save and cache restore operations only take a couple of seconds each.
The cache content gets "left behind" at the end of the run. Should we add a step that at least tries to delete the cached file(s) at the end of the run?
(I see that upload-coverage has a purge-cache step to cleanup its crud https://drone.owncloud.com/owncloud/ocis/1808/14/5 )

File: .drone.star [hunk 3]
```diff
@@ -1620,7 +1620,7 @@ def genericCachePurge(name, cache_key):
   return {
     'kind': 'pipeline',
     'type': 'docker',
-    'name': 'purge_%s_build_artifact_cache' %(name),
+    'name': 'purge_%s' %(name),
```
Comment (wkloucek) [inline .drone.star:1623]: Long pipeline names like "purge_ocis-binary-amd64_build_artifact_cache_build_artifact_cache" seems to lead to a silent fail of drone pipeline....

Comment (phil-davis) [reply .drone.star:1623]: yes, there is an "unfortunate" 50-character limit!

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/ocis/1838/81/2
purge-cache gets an error. I guess the cache needs to be "restored" first to that pipeline, then the folder and files will be visible and can be deleted.

**Comment by wkloucek** [comment]
> https://drone.owncloud.com/owncloud/ocis/1838/81/2
> purge-cache gets an error. I guess the cache needs to be "restored" first to that pipeline, then the folder and files will be visible and can be deleted.

Fixed: https://drone.owncloud.com/owncloud/ocis/1867/81/2
I used the minio cli the wrong way...

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/ocis/1867/81/2
cache purge worked:
```
+ mc rm --recursive --force cache/cache/ocis/owncloud/ocis/2394de10f79f60ce8c551ffe1399214ea262477e-1867/ocis-binary-amd64_build_artifact_cache
Removing `cache/cache/ocis/owncloud/ocis/2394de10f79f60ce8c551ffe1399214ea262477e-1867/ocis-binary-amd64_build_artifact_cache/ocis/bin/ocis`.
```

**Pull request merged**